lib/logmsg/nvtable.c

```diff
@@ -718,7 +718,7 @@ nv_table_realloc(NVTable *self, NVTable **new)
   if (new_size == old_size)
     return FALSE;
 
-  if (self->ref_cnt == 1 && !self->borrowed)
+  if (g_atomic_counter_get(&self->ref_cnt) == 1 && !self->borrowed)
```

Comment:
Is it possible that more thread runs this code path concurrently?

Reply:
nv_table_realloc is called when the message is modified: new value pairs are added or removed. Sources and destinations do not share logmessage/nvtable, the rest of the logpipes are protected by logmessage write protection and COW. So this line is safe.

Borrowed is only changed during `nv_table_realloc`, and `nv_table_clone`. `nv_table_clone` is also called only when the message is changed. So borrowed does not need to be protected against concurrent access.

Reply:
okay, thanks.